### PR TITLE
Iterate the docs of ActiveRecord::Transaction

### PR DESCRIPTION
**Pull request by fxn**
The purpose of this patch is to define upfront the concepts involved in this API.

@casperisfine what do you think?

### CODE REVIEW

File: activerecord/lib/active_record/transaction.rb [hunk 1]
```diff
@@ -5,16 +5,29 @@
 module ActiveRecord
   # Class specifies the interface to interact with the current transaction state.
   #
-  # It can either map to an actual transaction or represent the abscence of a transaction.
+  # It can either map to an actual transaction/savepoint, or represent the
+  # abscence of a transaction.
```
Comment (casperisfine) [inline activerecord/lib/active_record/transaction.rb:9]:
```suggestion
  # absence of a transaction.
```
That one is on me...

**Review by casperisfine — COMMENTED** [review]
LGTM

**Comment by fxn** [comment]
@casperisfine just pushed an iteration over

> If there is no currently open transactions, the block is called immediately.

because if the transaction is closed and materialized, it is not open, and it does call the block. At the bottom we had the exception, basically I put the two of them together.

**Comment by fxn** [comment]
Backported to `7-2-stable` 👍.

**Pull request merged**